experimental/Schemes/SpaceGerms.jl

```diff
@@ -0,0 +1,233 @@
+                                            <:MPolyComplementOfKPointIdeal}
+                        }
+GermAtGeometricPoint = Spec{<:Ring, 
+                            <:AbsLocalizedRing{<:Ring, <:Any, 
+                                               <:MPolyComplementOfPrimeIdeal}
+                           }
+
+###################################################################
+## start of the definition of space germ functionality
+###################################################################
+
+@Markdown.doc """
+    SpaceGerm{BaseRingType, RingType, SpecType}
+A space germ ``(X,O_{(X,x)}``, i.e. a ringed space with underlying scheme ``X`` of type SpecType and local ring ``O_{(X,x)}`` of type `RingType` over some base ring ``k`` of type `BaseRingType`.
+"""
+@attributes mutable struct SpaceGerm{BaseRingType<:Ring, RingType<:Ring, SpecType<:Spec} <: AbsSpaceGerm{BaseRingType, RingType}
+  X::SpecType
+
+  function SpaceGerm(X::GermAtClosedPoint)
+    return new{typeof(base_ring(X)), typeof(OO(X)), typeof(X)}(X)
+  end
+  
+  function SpaceGerm(X::GermAtGeometricPoint)
+    return new{typeof(base_ring(X)), typeof(OO(X)), typeof(X)}(X)
+  end
+end
+
+### Getter functions
+
+function underlying_scheme(X::SpaceGerm)
+  return X.X
+end
+
+@attr Spec function representative(X::SpaceGerm{<:Ring, <:MPolyQuoLocalizedRing})
+    R = base_ring(OO(X))
+    I = modulus(OO(X))
+    Q, _ = quo(R, I)
+    return Spec(Q)
+end
+
+@attr Spec function representative(X::SpaceGerm{<:Ring, <:MPolyLocalizedRing})
+    R = base_ring(OO(X))
+    return Spec(R)
+end
+
+function point(X::SpaceGerm{<:Any, <:Any, <:GermAtClosedPoint})
+  return point_coordinates(inverted_set(OO(X)))
+end
+
+function point(X::SpaceGerm{<:Any, <:Any, <:GermAtGeometricPoint})
+  return prime_ideal(inverted_set(OO(X)))
+end
+
+ring_of_germ(X::AbsSpaceGerm) = OO(X)
+
+function ideal_of_germ(X::AbsSpaceGerm{<:Ring,<:MPolyQuoLocalizedRing})
+    return localized_ring(OO(X))(modulus(OO(X)))
+end
+
+function ideal_of_germ(X::AbsSpaceGerm{<:Ring,<:MPolyLocalizedRing})
+    return ideal(OO(X),[zero(OO(X))])
+end
+
+function ambient_germ(X::AbsSpaceGerm{<:Ring,<:MPolyQuoLocalizedRing})
+    Y,_ = germ_at_point(localized_ring(OO(X)))
+    return Y
+end
+
+function ambient_germ(X::AbsSpaceGerm{<:Ring,<:MPolyLocalizedRing})
+    return X
+end
+
+############################################################################################################
+# allow user to specify point also as ideal
+# currently broken -- suggestions welcome on how to do this Oscar-style
+############################################################################################################
+function _maxideal_to_point(I::MPolyIdeal)
+  G = groebner_assure(I)
+  dim(I)==0 || error("Ideal does not describe finite set of points")
+  singular_assure(G)
+  vdim(G)==1 || error("Ideal does not describe a single K-point")
+  return [singular.reduce(v,G) for v in gens(base_ring(I))]
```

Comment:
This looks like trying to bring Oscar polynomials to normal forms using a Singular gb. I think the command you are looking for on the Oscar side is `normal_form`.

Reply:
Yes, this is what I war looking for...